### PR TITLE
Update Doc string for SnowflakeOperatorAsync

### PR DESCRIPTION
**Pull request by bharanidharan14**
Added Doc string for the SnowflakeOperatorAsync to get to know the use of this operator compared with snowflake sql API operator, best practices to be followed while using this operator and where can this `SnowflakeOperatorAsync` can be used

closes: #557
Screenshot 2022-07-28 at 2 56 48 PM
Screenshot 2022-07-28 at 5 56 44 PM

### CODE REVIEW

**Comment by bharanidharan14** [comment]
@phanikumv  @rajaths010494 @pankajastro @pankajkoti, can you review this PR

**Comment by pankajastro** [comment]
> @phanikumv @rajaths010494 @pankajastro @pankajkoti, can you review this PR

Can you please post the screenshot like how it is looking you can run `make docs`?

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 1]
```diff
@@ -19,7 +19,38 @@
...
+    - submit multiple queries in parallel without waiting for each query to complete.
+    - Accept List of query or  multiple query with ‘;’ semicolon separated string and params
+    - It loops through the query and execute the query in sequence order and uses execute_async method to run the query
+    - Once the query is submitted it executes the query from one connection and get the query id from the
+    response and passed it trigger and close the connection. so that the worker slots are made free.
+    - The trigger gets the query id list as input and poll every few seconds to snowflake and check
+    for the query status based on the query id from different connection.
+
+    .. see also::
+        https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.
```
Comment (pankajastro) [inline astronomer/providers/snowflake/operators/snowflake.py:33]: see also directive in between docs string creates a weird big block.

**Comment by rajaths010494** [comment]
LGTM , once Screenshot is added we can close.

**Comment by bharanidharan14** [comment]
@rajaths010494 @pankajastro Added screenshot

**Comment by pankajastro** [comment]
> @rajaths010494 @pankajastro Added screenshot

I can't see the below text in the screenshot but it is in the doc string. I was curious to see how it looks otherwise changes look good to me 
> for the query status based on the query id from different connection.
    .. see also::
        https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.

**Comment by bharanidharan14** [comment]
> > @rajaths010494 @pankajastro Added screenshot
> 
> I can't see the below text in the screenshot but it is in the doc string. I was curious to see how it looks otherwise changes look good to me
> 
> > for the query status based on the query id from different connection.
> > .. see also::
> > https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.

See also link is not visible in Docs, its not showing up in any of the operators/sensors

**Comment by bharanidharan14** [comment]
> > > > @rajaths010494 @pankajastro Added screenshot
> > > 
> > > 
> > > I can't see the below text in the screenshot but it is in the doc string. I was curious to see how it looks otherwise changes look good to me
> > > > for the query status based on the query id from different connection.
> > > > .. see also::
> > > > https://docs.snowflake.com/en/user-guide/python-connector-example.html#label-python-connector-querying-data-asynchronous.
> > 
> > 
> > See also link is not visible in Docs, its not showing up in any of the operators/sensors
> 
> then this is a bug in our documentation please create a ticket to track it. But I was able to do it in #464

Created Ticket

**Review by pankajastro — APPROVED** [review]

**Review by rajaths010494 — APPROVED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 2]
```diff
@@ -19,7 +19,35 @@
...
+      for asynchronous execution.
+    - submit multiple queries in parallel without waiting for each query to complete.
+    - Accept List of query or  multiple query with ‘;’ semicolon separated string and params It loops through the
+      query and execute the query in sequence order and uses execute_async method to run the query
+    - Once the query is submitted it executes the query from one connection and get the query id from the
+      response and passed it trigger and close the connection. so that the worker slots are made free.
+    - The trigger gets the query id list as input and poll every few seconds to snowflake and check
+      for the query status based on the query id from different connection.
+
+    where can this operator fit in?
```
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:32]:
```suggestion
    Where can this operator fit in?
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 3]
```diff
@@ -19,7 +19,35 @@
...
+         - Whether the single query or multiple query to execute the query in parallel and free up the worker resource.
```
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:33]:
```suggestion
```

Comment (phanikumv) [reply astronomer/providers/snowflake/operators/snowflake.py:33]: Remove above as it is already covered in previous point

**Review by phanikumv — CHANGES_REQUESTED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 3 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:33]: Removed this line
`Whether the single query or multiple query to execute the query in parallel and free up the worker resource.`

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 2 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:32]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 4]
```diff
@@ -19,7 +19,35 @@
 
 class SnowflakeOperatorAsync(SnowflakeOperator):
     """
-    Executes SQL code in a Snowflake database
+    - SnowflakeAsyncOperator uses the snowflake python connector ``execute_async`` method to submit a database command
+      for asynchronous execution.
+    - submit multiple queries in parallel without waiting for each query to complete.
```
[the remark this answers is not in the document]
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:24]: Changed the first letter to caps

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 5]
```diff
@@ -19,7 +19,35 @@
...
+    - Accept List of query or  multiple query with ‘;’ semicolon separated string and params It loops through the
+      query and execute the query in sequence order and uses execute_async method to run the query
```
[the remark this answers is not in the document]
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Addressed as per suggested comment

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 6]
```diff
@@ -19,7 +19,35 @@
...
+    - Once the query is submitted it executes the query from one connection and get the query id from the
```
[the remark this answers is not in the document]
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Fixed

**Comment by bharanidharan14** [comment]
@phanikumv Addressed your PR review changes

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 7]
```diff
@@ -128,8 +155,9 @@ class SnowflakeSqlApiOperatorAsync(SnowflakeOperator):
     This Operator currently uses key pair authentication, so you need tp provide private key raw content or
     private key file path in the snowflake connection along with other details
 
-    .. see also::
-        https://docs.snowflake.com/en/developer-guide/sql-api/authenticating.html#label-sql-api-authenticating-key-pair
+    .. seealso::
+
+        `Snowflake SQL API key pair Authentication <https://docs.snowflake.com/en/developer-guide/sql-api/authenticating.html#label-sql-api-authenticating-key-pair>`_  # noqa
 
     where can this operator fit in?
```
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:162]:
```suggestion
    Where can this operator fit in?
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 8]
```diff
@@ -128,8 +155,9 @@ class SnowflakeSqlApiOperatorAsync(SnowflakeOperator):
...
          - To Execute Multiple SQL statement in single request
```
Comment (phanikumv) [inline astronomer/providers/snowflake/operators/snowflake.py:163]:
```suggestion
         - To execute multiple SQL statements in a single request
```

**Review by phanikumv — CHANGES_REQUESTED** [review]

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 8 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:163]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 7 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:162]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 9]
```diff
@@ -19,7 +19,34 @@
 
 class SnowflakeOperatorAsync(SnowflakeOperator):
     """
-    Executes SQL code in a Snowflake database
+    - SnowflakeAsyncOperator uses the snowflake python connector ``execute_async`` method to submit a database command
```
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:22]:
```suggestion
    - SnowflakeOperatorAsync uses the snowflake python connector ``execute_async`` method to submit a database command
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 10]
```diff
@@ -19,7 +19,34 @@
...
+      for asynchronous execution.
+    - Submit multiple queries in parallel without waiting for each query to complete.
+    - Accept List of queries or multiple queries with ‘;’ semicolon separated string and params. It loops through the
```
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:25]:
```suggestion
    - Accepts list of queries or multiple queries with ‘;’ semicolon separated string and params. It loops through the
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 11]
```diff
@@ -19,7 +19,34 @@
...
+      queries and execute the query in sequence. Uses execute_async method to run the query
```
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:26]:
```suggestion
      queries and executes them in sequence. Uses execute_async method to run the query
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 12]
```diff
@@ -19,7 +19,34 @@
...
+    - Once a query is submitted, it executes the query from one connection and gets the query id from the
+      response and passes it to the Triggerer and closes the connection (so that the worker slots can be freed up).
+    - The trigger gets the query id list as input and poll every few seconds to snowflake and check
```
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:29]:
```suggestion
    - The trigger gets the list of query IDs as input and polls every few seconds to snowflake and checks
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 13]
```diff
@@ -19,7 +19,34 @@
...
+      for the query status based on the query id from different connection.
```
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:30]:
```suggestion
      for the query status based on the query ID from different connection.
```

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 14]
```diff
@@ -19,7 +19,34 @@
...
+
```
Comment (pankajkoti) [inline astronomer/providers/snowflake/operators/snowflake.py:31]: Do we need `-` for the above statements? I guess we can remove those.

**Review by pankajkoti — APPROVED** [review]
LGTM, minor suggestions.

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 14 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:31]: Document doesn't look good Without `-`, Initially I added as paragraph it looks congested

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 13 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:30]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 12 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:29]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py [hunk 15]
```diff
@@ -19,7 +19,34 @@
 
 class SnowflakeOperatorAsync(SnowflakeOperator):
     """
-    Executes SQL code in a Snowflake database
+    - SnowflakeAsyncOperator uses the snowflake python connector ``execute_async`` method to submit a database command
+      for asynchronous execution.
+    - Submit multiple queries in parallel without waiting for each query to complete.
+    - Accept List of queries or multiple queries with ‘;’ semicolon separated string and params. It loops through the
+      queries and execute the query in sequence. Uses execute_async method to run the query
+    - Once a query is submitted, it executes the query from one connection and gets the query id from the
```
[the remark this answers is not in the document]
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:27]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 11 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py:26]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 10 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Fixed

File: astronomer/providers/snowflake/operators/snowflake.py (hunk 9 above)
Comment (bharanidharan14) [reply astronomer/providers/snowflake/operators/snowflake.py]: Fixed

**Comment by bharanidharan14** [comment]
@phanikumv Addressed your review comments, can you please review it again

**Comment by bharanidharan14** [comment]
@phanikumv Need your approval

**Review by phanikumv — APPROVED** [review]

**Pull request merged**